uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -98,12 +98,11 @@ public static function validateParams($params, $validationMode = self::VALIDATIO
     self::validateLeaveDatesDoesNotOverlapContractsWithLapses($params);
     self::validateAbsenceTypeIsActiveAndValid($params, $absenceType);
     self::validateTOILRequest($params, $absenceType, $absencePeriod);
-    self::validateLeaveDaysAgainstAbsenceTypeMaxConsecutiveLeaveDays($params, $absenceType);
     self::validateAbsenceTypeAllowRequestCancellationForLeaveRequestCancellation($params, $absenceType);
     self::validateAbsencePeriod($params, $absencePeriod);
 
     if($validationMode != self::IMPORT_VALIDATION) {
-      self::validateEntitlementAndWorkingDayAndBalanceChange($params, $absenceType, $absencePeriod);
+      self::validateMaxConsecutiveDaysAndEntitlementAndWorkingDayAndBalanceChange($params, $absenceType, $absencePeriod);
```

Comment:
This doesn't look good. What if we need a new validation that also depends on balance changes or work days? Will we just keep increasing this method name? The main point of my other comment was: "I wonder if we can do this without calculateBalanceChange()", but we are still calling that method. The only difference is that now we call it only once.

The only validation that really needs the balance change is the entitlement one (where we check if the request balance is not greater than the remaining balance). Everything else has nothing to do with that. For the Working Days and the Consecutive Days validations, all we need is the information from the Work Pattern. So, when I was talking about splitting things, I was talking about splitting those two types of validations.